workspaces/optic-engine/src/projections/endpoint.rs

```diff
@@ -129,6 +130,33 @@ impl EndpointProjection {
     self.node_id_to_index.insert(request_id, request_node_index);
   }
 
+  pub fn without_request(&mut self, request_id: RequestId) {
+    let request_node_index = *self
+      .node_id_to_index
+      .get(&request_id)
+      .expect("expected request_id to have a corresponding node");
+
+    let method_parent_edge_index = self
+      .graph
+      .edges_directed(request_node_index, petgraph::Direction::Outgoing)
+      .find(|parent_edge| {
+        let parent_node_index = parent_edge.target();
+        let node = self.graph.node_weight(parent_node_index);
+        matches!(node, Some(Node::HttpMethod(_)))
+      })
+      .map(|parent_edge| parent_edge.id());
+
+    if let Some(method_parent_edge_index) = method_parent_edge_index {
+      self.graph.remove_edge(method_parent_edge_index); // prevents request to be resolved from path node
+    }
+    self.node_id_to_index.remove(&request_id); // prevents request node to be looked up by request id
+
+    // GOTCHA: we're not deleting the request node itself, as that would invalidate self.node_id_to_index
+    // as the graph indexes shift.
+    // TODO: figure out the implications of the above, like correctness of queries and
+    // eventual garbage collection.
+  }
```

Comment:
@devdoshi would value your thoughts on this approach and the resulting gotchas. What this does is:

- removes the edge between the `HttpMethod` node and `Request` node. This should prevent the queries to resolve this request, but allows Response to still exist (to be removed separately through its own event, even though it's technically possible to not issue those, leaving responses without a request).
- removes the request id from the node id to node index map. Should prevent queries that try to resolve the node per index. Does mean that technically these ids could be re-used, which I'm not 100% sure would have any negative implications. We could keep a separate list of deleted ids to prevent this.
- leaves all the actual graph nodes in place, as removing them will require the `node_id_to_index` to be rebuilt, which seems non-trivial. That does mean we have orphaned nodes hanging around. Probably not immediately an issue, but something we'd want to keep tabs on.